library/core/src/mem/maybe_uninit.rs

```diff
@@ -1321,3 +1305,29 @@ impl<T, const N: usize> [MaybeUninit<T>; N] {
         unsafe { super::transmute_copy(&ManuallyDrop::new(self)) }
     }
 }
+
+// --- FIXME ---
+// Stabilize these as From implementations.
+// The only reason they aren't From is because that's insta-stable.
```

Comment:
IMO `into` is *way* to implicit for use around unsafe code. Unsafe code needs clearly stated intent, so just saying "please convert this to something else" is not explicit enough.

So IMO we actually want dedicated methods here, and we need to bikeshed their name. I don't like `inner` but I also don't have better suggestions at the moment.

Reply:
I went with `into_inner` to be consistent with other APIs.

Reply:
I don't know any API this is consistent with? `into_inner` is usually for methods of the form `Type<T> -> T`; I find it extremely surprising to see it here for a raw pointer method.

Something like `as_inner_ptr` or so maybe?

Reply:
I guess I don't see `*const Type<T> -> *const T` as being too different from `Type<T> -> T`. I guess the difference is that one is a type cast whereas the other is providing a view into a type.

`as_` is supposed to be for borrowed types—are we counting pointers as a kind of borrow for naming purposes? If so `as_inner_ptr` seems great or maybe even just `as_inner`.

Reply:
> I don't see *const Type<T> -> *const T as being too different from Type<T> -> T

Why not? It makes a huge difference whether ownership is consume vs whether there is a form of "borrowing". We never use `into_inner` for methods like `&Type<T> -> &T` or `&mut Type<T> -> &mut T`. These functions here are very much like them, except they are borrowing via raw pointers, not references.

One similar function we have is [`UnsafeCell::raw_get`](https://doc.rust-lang.org/nightly/core/cell/struct.UnsafeCell.html#method.raw_get). On `MaybeUninit` we already have `as_(mut_)ptr` that take references; now here we have the exact same methods using raw pointers so the naming should be similar. `raw_as_ptr` sounds odd though...

Reply:
Gotya, I wasn't really thinking of pointers as references for naming but you're totally right that makes more sense. `raw_as_ptr` does seems a little weird, but it would be consistent... will rename.